app/views/course_management/available_courses/show.html.erb

```diff
@@ -2,7 +2,9 @@
   <div class='col-lg-12'>
     <div class='alert alert-light'>
       <%= link_to_back available_courses_path %>
-      <%= link_to_actions(@available_course, except: :show) %>
+      <% manageable = @available_course.manageable? %>
+      <% except = %i[show] << (:destroy unless manageable) %>
+      <%= link_to_actions(@available_course, except: except.compact) %>
```

Comment:
Burasıda biraz karışık olmuş.

Aşağıdaki gibi düzenleyebiliriz. Test etmedim.

```suggestion
      <% manageable = @available_course.manageable? %>
      <% if  @available_course.manageable? %>
          <%= link_to_actions(@available_course, except: :show) %>
      <% else %>
           <%= link_to_edit([:edit, @available_course]) %>
      <% end %>
```

İşin en temeline baktığımızda bu bir `authorization` problemi yani https://github.com/omu/nokul/blob/develop/app/policies/course_management/available_course_policy.rb dosyasında`edit, update, destroy` metodları için kısıtlama eklenmesi lazım. Şuan link'lerin görüntülenme durumları izinlere göre değişmiyor ama bir sonraki yapacağımız geliştirme ile buda otomatik olacak ve bu tür koşullar gerekmeyecek. Dediğim özellik gelene kadar view tarafında da policy düzenlendikten sonra `policy(@available_course).edit?` koşuluna göre görünüm kısıtlaması yapabilirsin.

Reply:
@isubas ile şimdilik eventler ilgili kontrollerin policy'e eklenmemesine karar verildi.